DelvUI/Interface/Jobs/DancerHud.cs

```diff
@@ -289,89 +289,44 @@ private void DrawStandardBar(Vector2 origin)
-            var windmillDuration = flourishingWindmillBuff.Duration;
-            var windmillStart = timersEnabled ? windmillDuration : 20;
-            var windmillPos = new Vector2(
-                origin.X + Config.Position.X + Config.ProcBarPosition.X + (2 * order[2] - 2) * Config.ProcBarSize.X / 4f - order[2] * procBarSize.X,
-                origin.Y + Config.Position.Y + Config.ProcBarPosition.Y - Config.ProcBarPosition.Y / 2f
-            );
-            var windmillBuilder = BarBuilder.Create(windmillPos, procBarSize)
-                .SetBackgroundColor(EmptyColor.Base);
-
-            if (windmillDuration > 0)
-            {
-                windmillBuilder.AddInnerBar(windmillStart, 20f, Config.FlourishingWindmillColor);
-            }
-
-            // Flourishing Shower
-            var flourishingShowerBuff = player.StatusEffects.FirstOrDefault(o => o.EffectId == 1817);
-            var showerDuration = flourishingShowerBuff.Duration;
-            var showerStart = timersEnabled ? showerDuration : 20;
-            var showerPos = new Vector2(
-                origin.X + Config.Position.X + Config.ProcBarPosition.X + (2 * order[3] - 2) * Config.ProcBarSize.X / 4f - order[3] * procBarSize.X,
-                origin.Y + Config.Position.Y + Config.ProcBarPosition.Y - Config.ProcBarPosition.Y / 2f
-            );
-            var showerBuilder = BarBuilder.Create(showerPos, procBarSize)
-                .SetBackgroundColor(EmptyColor.Base);
-
-            if (showerDuration > 0)
-            {
-                showerBuilder.AddInnerBar(showerStart, 20f, Config.FlourishingShowerColor);
-            }
+            var order = Config.procsOrder;
+            var procTimers = new float[] {cascadeTimer, fountainTimer, windmillTimer, showerTimer};
+            var colors = new PluginConfigColor[] { Config.FlourishingCascadeColor, Config.FlourishingFountainColor, Config.FlourishingWindmillColor, Config.FlourishingShowerColor};
 
-            if (!Config.StaticProcBarsEnabled)
+            for(int i = 0; i < 4; i++)
             {
-                cascadeBuilder.SetTextMode(BarTextMode.Single).SetText(BarTextPosition.CenterMiddle, BarTextType.Current);
-                fountainBuilder.SetTextMode(BarTextMode.Single).SetText(BarTextPosition.CenterMiddle, BarTextType.Current);
-                windmillBuilder.SetTextMode(BarTextMode.Single).SetText(BarTextPosition.CenterMiddle, BarTextType.Current);
-                showerBuilder.SetTextMode(BarTextMode.Single).SetText(BarTextPosition.CenterMiddle, BarTextType.Current);
+                if(Config.StaticProcBarsEnabled)
+                {
+                    var builder = BarBuilder.Create(cursorPos, procBarSize).
+                        AddInnerBar(procTimers[order[i]] > 0 ? 1 : 0, 1, colors[order[i]]);
+                    builder.Build().Draw(drawList);
+                }
+                else
+                {
+                    var builder = BarBuilder.Create(cursorPos, procBarSize).
```

Comment:
Just insert .SetBackgroundColor(EmptyColor.Base) on the builder variables so it doesn't look out of place and I'd be happy with all of this, see my other comment